build-release-tools/application/revert_submodule.py

```diff
@@ -0,0 +1,137 @@
+                        action="append")
+
+
+
+    parsed_args = parser.parse_args(args)
+    return parsed_args
+
+def get_current_submodule_list(submodules,repo_dir):
+    """
+    Parse the repo url to get the repo folder name
+    """
+    submodule_folders = []
+    if len(submodules) == 0:
+        return None
+    try:
+       submodules_list = submodules.split('\n')
+       for item in submodules_list :
+           if len(item)!= 0:
+              submodule_item_list = item.split()
+              submodule_folders.append(submodule_item_list[1])
+    except KeyError as error:
+        print "Fail to process the submodules of this repo name {0}".format(repo_dir)
+    return submodule_folders
+       
+def get_previous_id(commit_info) :
+    """
+    Function is used to get the last commit id of update submodule
+    """
+    try:
+        commit_info_list = commit_info.split('\n')
+        item = commit_info_list[0]
+        if len(item) != 0:
+            item_list = item.split()
+            return item_list[1]
+
+    except Exception,e:
+        return None
+def main():
+    # parse arguments
+    args = parse_command_line(sys.argv[1:])
+    if args.publish:
+        if args.git_credential:
+            repo_operator = RepoOperator(args.git_credential)
+        else:
+            print "Error occurs when get crendtail in update submodule"
+            sys.exit(1)
+
+    if os.path.isdir(args.build_dir):
+        print args.build_dir
+        for filename in os.listdir(args.build_dir):
+            try:
+                repo_dir = os.path.join(args.build_dir, filename)
+                repo_operator.submodule_init(repo_dir)
+                repo_operator.submodule_update(repo_dir)
+
+                submodules_list = repo_operator.get_current_submodule(repo_dir)
+                if len(submodules_list)==0:
+                    continue;
+                #git pull is used to get the lastest code to make sure push successfully
+                test_info = repo_operator.git_pull(repo_dir)
+                update_commit_message="update submodule for new commit {0}".format(args.version)
+                commit_info = repo_operator.get_commit_of_update_submodule(repo_dir,update_commit_message)
+                commit_id = get_previous_id(commit_info)
+                if commit_id != None:
+                        repo_operator.revert_to_commit(repo_dir,commit_id)
+                if args.publish:
+                    print "start to publish  revert update submodule in {0}".format(repo_dir)
+                    commit_message = "revert update submodule for new commit {}".format(args.version)
+                    repo_operator.push_repo_changes(repo_dir, commit_message)
+            except Exception,e:
+                print "Failed to revert update submodule of {0} due to {1}".format(filename, e)
+                sys.exit(1)
+    else:
+        print "The argument build-dir must be a directory"
+        sys.exit(1)
```

Comment:
Could you please refactor line 117~133 to a method like: revertCommitForUpdateSubModules?